### ISSUE TITLE
inconsisty between semantics in abstract gringo and implementation

### ISSUE THREAD

**Issue by rkaminsk** [potassco/clingo#23]
At the moment, the following rule has a different semantics in the gringo implementation and abstract gringo:
````
:- { e(X,Y) } >= 1, n(X;Y).
````
According to AG both `X` and `Y` have to be global variables.

**Comment by rkaminsk** [comment]
Working on this also uncovered a regression in clingo-5. The program
````
q(1..2).
{ p(X) : q(X) } :- 2 { q(X) }.
````
produces wrong results with clingo-5 because variable `X` wrongly became global.

**Comment by rkaminsk** [comment]
Working on this also uncovered a second regression in clingo-5. The program
````
x :- #count { 1 : p(X+Y) } >= 0, X=1, Y=a.
````
produces wrong results with clingo-5 because `X+Y` is wrongly pulled into the rule body (clingo-4 guarantees that a term always has a value).

**Comment by rkaminsk** [comment]
Fixed with d0283b8365d79f052c5eece4de47510e8a5c3855
